### PR TITLE
futher simplify remove and shift_back operations in hashmap

### PR DESCRIPTION
**Pull request by bobzhang**
- simplify the remove operation
- simplify shift_back
- tweak

### CODE REVIEW

File: hashmap/hashmap.mbt [hunk 1]
```diff
@@ -135,9 +135,7 @@ pub fn contains[K : Hash + Eq, V](self : HashMap[K, V], key : K) -> Bool {
 /// Remove a key-value pair from hash map.
 pub fn remove[K : Hash + Eq, V](self : HashMap[K, V], key : K) -> Unit {
   let hash = key.hash()
-  for i = 0, idx = hash.land(self.capacity - 1)
-      i < self.capacity
```
Comment (bobzhang) [inline hashmap/hashmap.mbt:139]: `i < self.capacity` check is not needed?

File: hashmap/hashmap.mbt [hunk 2]
```diff
@@ -147,37 +145,37 @@ pub fn remove[K : Hash + Eq, V](self : HashMap[K, V], key : K) -> Unit {
...
       }
-      None => ()
+      None => break
     }
   }
 }
 
 fn shift_back[K : Hash, V](self : HashMap[K, V], start_index : Int) -> Unit {
-  for i = 0, prev = start_index, curr = (start_index + 1).land(
-          self.capacity - 1,
```
Comment (bobzhang) [inline hashmap/hashmap.mbt:160]: the loop state variable `i` is not needed

**Comment by bobzhang** [comment]
@coderabbitai review

**Review by yj-qin — APPROVED** [review]

**Comment by yj-qin** [comment]
The 50% load factor will result in half of the space being wasted, maybe a higher load factor is better?

**Comment by bobzhang** [comment]
> The 50% load factor will result in half of the space being wasted, maybe a higher load factor is better?

I am open to change the load factor later, rationale why I prefer 0.5 instead of higher values:
1. when people use hashtbl, they want the optimal performance, so it is okay to exchange space for speed
2. the wasted the space is not that bad, since the array only holds a pointer(not the full content)

**Pull request merged**